From 0cd153b2681437b91f07315a27238446cc24363f Mon Sep 17 00:00:00 2001
From: Timur Pocheptsov <timur.pocheptsov@qt.io>
Date: Thu, 12 Jan 2023 10:44:34 +0100
Subject: [PATCH] QWidgetTextControl: deal with empty formats list properly

Do not call first on it. A regression was introduced by
56f0ebfe860e440dcbba8997f44836debc901119.

Fixes: QTBUG-110070
Pick-to: 6.5 6.4
Change-Id: I7c969da889b6c7ad3a67149fe99ca47830683073
Reviewed-by: Shawn Rutledge <shawn.rutledge@qt.io>
---
 src/widgets/widgets/qwidgettextcontrol.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/widgets/widgets/qwidgettextcontrol.cpp b/src/widgets/widgets/qwidgettextcontrol.cpp
index 1b78442ed1c..4f008fe24f7 100644
--- a/src/widgets/widgets/qwidgettextcontrol.cpp
+++ b/src/widgets/widgets/qwidgettextcontrol.cpp
@@ -2699,7 +2699,8 @@ void QWidgetTextControl::insertFromMimeData(const QMimeData *source)
     bool hasData = false;
     QTextDocumentFragment fragment;
 #if QT_CONFIG(textmarkdownreader)
-    if (source->formats().first() == "text/markdown"_L1) {
+    const auto formats = source->formats();
+    if (formats.size() && formats.first() == "text/markdown"_L1) {
         auto s = QString::fromUtf8(source->data("text/markdown"_L1));
         fragment = QTextDocumentFragment::fromMarkdown(s);
         hasData = true;
